### ISSUE TITLE
the cell index is not consequent when create tiles

### ISSUE THREAD

**Issue by alexzheng** [godotengine/godot#34325]
**Godot version:3.1.2

**OS/device including version:**

**Issue description:**
When create tile use the new single tile button, the cell index is not consequent after delete some tiles.
**Steps to reproduce:**
1. add some new tiles
2. delete some tiles
3. add some new tiles

the cell index of the tile is not consequent, the index of deleted cell will not used.

**Minimal reproduction project:**
<img width="1174" alt="Screen Shot 2019-12-13 at 5 43 21 PM" src="https://user-images.githubusercontent.com/1130047/70790842-d4dedd80-1dd0-11ea-9b3d-8d22ef3fd2f1.png">

**Comment by Zylann** [comment]
Changing the way tile indexes are generated can have effects on existing tilemaps. If you delete a tile, it's normal that there are non-contiguous indexes, because if you shift the others down to fill the gap, it will destroy your tilemaps.

Now, knowing that there can be holes *anyways*, I wonder why you would want indexes to be re-used with new tiles.  What is the actual problem you are trying to solve?

**Comment by alexzheng** [comment]
it's quite easy to make a mistake when drag and select a region to make a tile, so delete it and select it again happened very frequently. 
So even before filling any tiles in the map, the tileset I got is full of holes. It's very ugly and not easy to check the range of index.

**Comment by alexzheng** [comment]
If I deselect a tile by mistake, it's possible to select it again as the same index, that really broke the code.

**Comment by Zylann** [comment]
It's interesting because up to now, I've always used the conversion workflow where you convert scenes with *named nodes* into tiles. Which means every tile had a name, and I never had to worry about IDs.
But if you make your tiles by barely selecting them around *and not giving them any name that you choose*, I feel using whatever ID the engine gives you is quite some trouble... I'm not really used to the new tiles editor tbh

**Comment by alexzheng** [comment]
I'm also used to the original way to make tileset.
I do not mean, when delete a tile, all the other tiles should be shift to keep the index consequent.
What I suggest is, for example, I have created tiles with index from 0 to 9, then I delete the tile 0 by mistake, I would like to select it again as tile 0, but now it is tile 10.

Maybe it would be better to let the  id configurable. Also take the above example,When I select the tile 0 again, the new id would be 10, but we can change it to 0 manually.

**Comment by Zylann** [comment]
Actually when taking a quick look I think you can still give a name to tiles:
![image](https://user-images.githubusercontent.com/1311555/70849723-4c1c7c00-1e7a-11ea-9c79-906d2bdcf35f.png)

**Comment by alexzheng** [comment]
Yes,  the name can be customized. But the name of the tile can not be get from a cell directly, it require an additional step by calling the tile_get_name in TileSet.
If the TileMap have api like get_cell_name(cell), the name of the tile can also to be used to determine which tile is in the selected cell.

**Comment by FabianoFabiouchi** [comment]
In my TileSet I have the tile_ids 1, 2, 4 and 5. It seems that when I use world_to_map and get_cellv, when the position is over a blank tile, it somehow returns with 3.

`func is_on_ladder() -> bool:
	var value = false
	var tile_id = tilemap.get_cellv(tilemap.world_to_map(position))
	
	if tile_id != TileMap.INVALID_CELL:
		print(tile_id) # -> prints 3
		if tilemap.tile_set.tile_get_name(tile_id) == "ladder":
			value = true

	return value`

My problem is, that 3 actually is not INVALID_CELL (-1), so I get lots of errors `Condition ' !tile_map.has(p_id) ' is true. returned: String()`

Any ideas how to fix?

**Comment by rdrpenguin04** [comment]
Is there any way to intentionally fill gaps in IDs by now? A project I'm working on bases physics on tile IDs, specifically acting as a heightmap, and it's frustrating to have to recreate from scratch every time an ID is off-by-one.

**Comment by rakkarage** [comment]
you can renumber and pack and reorder by editing this list at end of file... maybe bit easier then recreate from scratch


	[resource]
	sources/0 = SubResource("TileSetAtlasSource_bqlht")
	sources/1 = SubResource("TileSetAtlasSource_7tsy3")
	sources/2 = SubResource("TileSetAtlasSource_yrua3")
	sources/3 = SubResource("TileSetAtlasSource_0kgob")
	sources/4 = SubResource("TileSetAtlasSource_i7o2d")
	sources/5 = SubResource("TileSetAtlasSource_b32ox")
	sources/6 = SubResource("TileSetAtlasSource_fi2r4")
	sources/7 = SubResource("TileSetAtlasSource_ihee1")
	sources/8 = SubResource("TileSetAtlasSource_0tsdn")
	sources/9 = SubResource("TileSetAtlasSource_wl51w")
